packages/manager/modules/pci/src/projects/project/instances/instance/shelve/shelve.controller.js

```diff
@@ -0,0 +1,77 @@
+import get from 'lodash/get';
+import Instance from '../../../../../components/project/instance/instance.class';
+
+export default class PciInstanceShelveController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiCloudProjectInstance,
+    PciProjectsProjectInstanceService,
+    Poller,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+    this.PciProjectsProjectInstanceService = PciProjectsProjectInstanceService;
+    this.Poller = Poller;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+  }
+
+  startPollerForShelveInstance() {
+    const endPointUrl = `/cloud/project/${this.projectId}/instance/${this.instance.id}`;
+    this.Poller.poll(endPointUrl, null, {
+      interval: 3000,
+      successRule(instance) {
+        return new Instance(instance).isShelved();
+      },
+      namespace: 'cloud.project.instance.shelve',
+      notifyOnError: false,
+    }).then(() => {
+      if (this.isInstancesStateScope()) {
```

Comment:
As mentioned by @marie-j & @frenautvh I think we should stop the polling when leaving the page/section.
Here the polling is never stopped, but you have to detect if the customer is still on the page or not. 
If the customer is still on the page, the list will refresh.
If the customer is in another section you don't refresh the instances list; the result will be the same if you destroy the poller when leaving.

Reply:
Yes true @cbourgois, but i consider one another things.
Case: customer return to the instances or sub instances page and polling back process just finished
=> in this case also i will refresh to them page.

I impalement it like this for this scenario.
Sure, if customer change the universe the tasks will be killed.

Reply:
In the case the customer is not on the instance list you still load the list again (even if he never visits the page again), and if the customer browse instances list of another projects (shelve an instance on project A and move to project B), the instances list of the project B will be refreshed.

Reply:
I changed my implementation.
Thanks to take a look again please.